google/cloud/bigquery/schema.py

```diff
@@ -73,14 +74,18 @@ def __init__(
         name,
         field_type,
         mode="NULLABLE",
-        description=None,
+        description=_DEFAULT_VALUE,
         fields=(),
         policy_tags=None,
     ):
-        self._name = name
-        self._field_type = field_type
-        self._mode = mode
-        self._description = description
+        self._properties = {
+            "name": name,
+            "type": field_type,
+        }
+        if mode is not None:
+            self._properties["mode"] = mode.upper()
+        if description is not _DEFAULT_VALUE:
```

Comment:
@shollyman This is one of the key changes: we no longer set the resource value for "description" if it's not explicitly set.

We already omit `policy_tags` from the resource if none (though arguably it should get the same treatment so that someone can unset policy tags from Python)

Reply:
My default inclination would be for special handling for None values to happen at the places where it's significant, like when calling tables.update.  It's also the case that schema fields can't be manipulated individually, so perhaps I'm simply just not thinking this through properly.

Reply:
I called that out as a possibility in https://github.com/googleapis/python-bigquery/issues/558, but that'd require updating our field mask logic to support sub-fields, which gets into some hairy string parsing (perhaps not all that hairy, as it could be as simple as split on `'.'`, but is definitely a departure from what we've been doing).

Also, it might mean that we'd have to introduce a field mask to our load job methods. Based on the error message we're seeing, it sounds like it's possible to make updates to fields like policy tags from a load job.